src/components/vm/nics/nicAdd.tsx

```diff
@@ -73,275 +78,224 @@ function getRandomMac(vms: VM[]): string | undefined {
-                    mode: this.state.networkSourceMode,
+                    "": tas.source,
+                    mode: tas.mode,
                 };
             } else {
-                source = this.state.networkSource;
+                source = tas.source;
             }
             let backend = null;
-            if (this.state.networkType == "user" && vm.capabilities.interfaceBackends.includes("passt"))
+            if (tas.type == "user" && vm.capabilities.interfaceBackends.includes("passt"))
                 backend = "passt";
+            if (source == "$create") {
+                createDefaultNetwork();
+                source = "default";
+            }
             await virtXmlHotAdd(
                 vm,
                 "network",
                 {
-                    mac: (this.state.setNetworkMac
-                        ? this.state.networkMac
-                        : getRandomMac(await this.props.appState.getVms())),
-                    model: this.state.networkModel,
-                    type: this.state.networkType,
+                    mac: (
+                        values.mac.set
+                            ? values.mac.val
+                            : getRandomMac(await appState.getVms())
+                    ),
+                    model: values.model,
+                    type: tas.type,
                     backend: { type: backend },
                     source,
-                    portForward: this.state.networkType == "user" ? dialogPortForwardsToInterface(this.state.portForwards) : null,
-
+                    portForward: (
+                        tas.type == "user"
+                            ? dialogPortForwardsToInterface(values.portForwards)
+                            : null
+                    ),
                 },
-                this.state.permanent
+                values.permanent
             );
             domainGet({ connectionName: vm.connectionName, id: vm.id });
-            Dialogs.close();
         } catch (exc) {
-            this.dialogErrorSet(_("Network interface settings could not be saved"), String(exc));
+            throw DialogError.fromError(_("Network interface settings could not be saved"), exc);
         }
-        this.setState({ addVNicInProgress: false });
     }
 
-    render() {
-        const Dialogs = this.context;
-        const { idPrefix, vm } = this.props;
+    const dlg = useDialogState(init, validate);
 
-        const defaultBody = (
-            <>
-                <Form onSubmit={e => e.preventDefault()} isHorizontal>
-                    <NetworkTypeAndSourceRow
-                        vm={vm}
-                        idPrefix={idPrefix}
-                        dialogValues={{ ...this.state, availableSources: this.props.availableSources }}
-                        onValueChanged={this.onBodyValueChanged}
-                    />
+    const defaultBody = (
+        <>
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2435-4e63f3e8-20251209-142520-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)